### PR TITLE
[CI] Set default Clang version to 10

### LINKED ISSUE #1441: Linux CI fails, compiler issue
First noticed in PR #1440 
https://github.com/trustwallet/wallet-core/runs/2640698260
and here, on master:
https://github.com/trustwallet/wallet-core/actions/runs/859471354

Seems to be related to change of default CLang version, from 10 to 11
https://github.com/trustwallet/wallet-core/runs/2640698260

Ubuntu v. used got changed from
https://github.com/actions/virtual-environments/blob/ubuntu20/20210510.0/images/linux/Ubuntu2004-README.md
to
https://github.com/actions/virtual-environments/blob/ubuntu20/20210517.1/images/linux/Ubuntu2004-README.md

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Fixes #1441 .

## Testing instructions

Linux CI build

## Types of changes

* Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

**Comment by hewigovens** [comment]
clang 11 will be default, let's use clang 11?

File: .github/workflows/linux-ci.yml [hunk 1]
```diff
@@ -15,6 +15,9 @@ jobs:
       run: |
         # build-essential libboost-all-dev clang-9 ruby-full cmake
```
Comment (hewigovens) [inline .github/workflows/linux-ci.yml:16]:
```suggestion
        # build-essential ruby-full cmake
```

**Comment by hewigovens** [comment]
```bash
libc++-dev is already the newest version (1:10.0-50~exp1).
libc++abi-dev is already the newest version (1:10.0-50~exp1).
```

can you remove this?

**Comment by catenocrypt** [comment]
Changed default clang version (back) to 10.
Better solution would be to use new 11, and fix compile issue.

**Review by hewigovens — APPROVED** [review]
don't forget to remove clang-9 from comment

**Review by ivan0117 — APPROVED** [review]

**Review by ivan0117 — APPROVED** [review]

**Review by ivan0117 — APPROVED** [review]

**Comment by ivan0117** [comment]
https://github.com/trustwallet/wallet-core/pull/1442#pullrequestreview-667589770

**Pull request merged**